warehouse/internal/repo/upload.go

```diff
@@ -377,6 +377,35 @@ func (uploads *Uploads) UploadJobsStats(ctx context.Context, destType string, op
 	return stats, nil
 }
 
+func (uploads *Uploads) UploadTimings(ctx context.Context, uploadID int64) (model.Timings, error) {
```

Comment:
To keep the interface of the upload repo a bit tight, would it be beneficial to look for the `upload_by_id` and fetch the information from it ?

Not sure, if that would be efficient. Depends on the call count maybe. Wdyt ?

Reply:
It is a good point, with some refactoring in the service layer it would make a lot of sense to retrieve the full model and modify it, then updated it in the db again.

I could simplify fetching the whole model here, but I am concerned with the risk of changing too much in this particular PR.

I can add a TODO and DEPRECATED comment here, so we don't forget them.

Once we're able to move all the DB access inside the repo, it would be much easier to refactor it into an ideal design.

Reply:
Sounds good to me 👍🏼